src/utils/cloud.ts

```diff
@@ -95,7 +95,7 @@ export const getDeviceAndMaybeAppFromUUID = _.memoize(
 	> => {
 		const device = await sdk.models.device.get(deviceUUID, deviceOptions);
 		const apps = device.belongs_to__application;
-		if (_.isEmpty(apps) || _.isEmpty(apps[0])) {
+		if (!apps.length || !apps[0]) {
```

Comment:
Can you make these checks better than purely `!`, I also suspect you don't need both checks and they were historically here because we didn't know if `device.belongs_to__application` would be a deferred object like `{ __id: 123 }` or an expanded array like `[{ id: 123 }]`

Reply:
Looks like we should just be able to remove this `if`, as an empty array's `apps[0]` should be `undefined` which means `return [device, apps[0]] as const;` will yield the same result as this `if`

Reply:
Sounds good